src/core/server/elasticsearch/version_check/ensure_es_version.ts

```diff
@@ -57,22 +58,28 @@ function getHumanizedNodeName(node: NodeInfo) {
 }
 
 export function mapNodesVersionCompatibility(
-  nodesInfo: NodesInfo,
+  nodesInfoResponse: NodesInfo & { nodesInfoRequestError?: Error },
   kibanaVersion: string,
   ignoreVersionMismatch: boolean
 ): NodesVersionCompatibility {
-  if (Object.keys(nodesInfo.nodes ?? {}).length === 0) {
+  if (Object.keys(nodesInfoResponse.nodes ?? {}).length === 0) {
+    // Note: If the a nodesInfoRequestError is present, the message contains the nodesInfoRequestError.message as a suffix
+    let message = `Unable to retrieve version information from Elasticsearch nodes.`;
+    if (nodesInfoResponse.nodesInfoRequestError) {
+      message = message + ` ${nodesInfoResponse.nodesInfoRequestError.message}`;
+    }
     return {
       isCompatible: false,
-      message: 'Unable to retrieve version information from Elasticsearch nodes.',
+      message,
       incompatibleNodes: [],
       warningNodes: [],
       kibanaVersion,
+      nodesInfoRequestError: nodesInfoResponse.nodesInfoRequestError, // optionally stringify the whole Error or parts of it that we're interested in
```

Comment:
I'm not sure I understand what does this comment mean? Are we planning to stringify the error in this PR?

Reply:
Oops, I forgot to delete that comment.